### PR TITLE
Replaced PhysicalConstants.jl with CLIMAParameters.

### PR DESCRIPTION
**Pull request by skandalaCLIMA**
However, there are a few places (src/rrtmgp/AtmosphericStates.jl) where different precisions are needed for some constants (grav, m_h2o, etc.).
In these cases, CLIMAParameters are overridden with local values.

### CODE REVIEW

File: src/rrtmgp/AtmosphericStates.jl [hunk 1]
```diff
@@ -35,6 +35,8 @@ computes column amounts of dry air using hydrostatic equation
 function get_col_dry(vmr_h2o, plev, tlay, latitude = nothing)
     FT = eltype(plev)
     ncol, nlay, nlev = size(plev, 1), size(tlay, 2), size(plev, 2)
+    m_dry = FT(0.028964) # higher precision needed here than CLIMAParameters
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:38]: Please grab this from CLIMAParameters and override the default values, as suggested in the [docs](https://clima.github.io/CLIMAParameters.jl/dev/).

File: src/rrtmgp/AtmosphericStates.jl [hunk 2]
```diff
@@ -35,6 +35,8 @@ computes column amounts of dry air using hydrostatic equation
 function get_col_dry(vmr_h2o, plev, tlay, latitude = nothing)
     FT = eltype(plev)
     ncol, nlay, nlev = size(plev, 1), size(tlay, 2), size(plev, 2)
+    m_dry = FT(0.028964) # higher precision needed here than CLIMAParameters 
+    m_h2o = FT(0.018016) # different precision needed here that CLIMAParameters
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:39]: Please grab this from CLIMAParameters and override the default values, as suggested in the [docs](https://clima.github.io/CLIMAParameters.jl/dev/).

Comment (charleskawczynski) [reply src/rrtmgp/AtmosphericStates.jl:39]: Also, please use the alias pattern that we've been following (`_grav = FT(grav(param_set))`)

File: src/rrtmgp/AtmosphericStates.jl [hunk 3]
```diff
@@ -46,18 +48,18 @@ function get_col_dry(vmr_h2o, plev, tlay, latitude = nothing)
     if latitude ≠ nothing
         g0 .= helmert1 - helmert2 * cos(FT(2) * π * latitude / FT(180)) # acceleration due to gravity [m/s^2]
     else
-        g0 .= grav(FT)
+        g0 .= helmert1
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:51]: Why has the name changed here? If this is actually gravity, please get this from CLIMAParameters.

File: test/PostProcessing.jl [hunk 4]
```diff
@@ -22,13 +25,13 @@ function compute_heating_rate(
...
-            flux_dn[:, ilay]
-        ) .* grav(FT) ./ (
-            cp_dry(FT) .* (as.p_lev[:, ilay+1] .- as.p_lev[:, ilay])
-        )
+    for ilay = 1:nlay
+        heating_rate[:, ilay] .=
+            (
+                flux_up[:, ilay+1] .- flux_up[:, ilay] .- flux_dn[:, ilay+1] .+
+                flux_dn[:, ilay]
+            ) .* grav(FT) ./
```
Comment (charleskawczynski) [inline test/PostProcessing.jl:33]:
```suggestion
            ) .* FT(grav(earth)) ./
```

File: src/rrtmgp/AtmosphericStates.jl (hunk 2 above)
Comment (charleskawczynski) [reply src/rrtmgp/AtmosphericStates.jl:39]: You'll need to pass the parameter set down the call graph to this function.

File: test/PostProcessing.jl (hunk 4 above)
Comment (charleskawczynski) [reply test/PostProcessing.jl:33]: Also, please create aliases using the pattern that we've been following (`_grav = FT(grav(param_set))`)

File: src/rrtmgp/AtmosphericStates.jl (hunk 3 above)
Comment (skandalaCLIMA) [reply src/rrtmgp/AtmosphericStates.jl:56]: These are the Helmert constants used by RRTMGP. I would avoid changing precision on just one of the two constants.

File: src/rrtmgp/AtmosphericStates.jl [hunk 5]
```diff
@@ -9,11 +9,13 @@ using DocStringExtensions
 
 using ..Utilities
 using ..MeshOrientations
-using ..PhysicalConstants
 using ..Gases
 using ..GasConcentrations
 using ..ReferenceStates
 
+using CLIMAParameters
+import CLIMAParameters.Planet: molmass_dryair, molmass_water
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:17]:
```suggestion
using CLIMAParameters.Planet: molmass_dryair, molmass_water
```
We should not be importing here, only `using`.

File: src/rrtmgp/AtmosphericStates.jl [hunk 6]
```diff
@@ -101,6 +107,8 @@ struct AtmosphericState{FT,I} <: AbstractAtmosphericState{FT,I}
     nlay::I
     "Number of columns."
     ncol::I
+    "Parameter set"
+    param_set::AbstractParameterSet
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:111]: This is fine for now, but if we want to make these runnable on the GPU, we'll need to make the type of `param` not abstract.

File: src/rrtmgp/AtmosphericStates.jl [hunk 7]
```diff
@@ -111,6 +119,7 @@ function AtmosphericState(
     ref::ReferenceState{FT},
     col_dry::Union{Array{FT,2},Nothing} = nothing,
     t_sfc::Union{Vector{FT},Nothing} = nothing,
+    param_set = EarthParameterSet(),
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:122]: This should be moved to the `test/` and passed down the call graph into `src/`

File: src/rrtmgp/AtmosphericStates.jl [hunk 8]
```diff
@@ -300,6 +310,7 @@ function Base.convert(
         mesh_orientation,
         nlay,
         ncol,
+        EarthParameterSet(),
```
Comment (charleskawczynski) [inline src/rrtmgp/AtmosphericStates.jl:313]: Again, this should be passed in as `param_set` from somewhere in `test/`

File: test/test_allsky.jl [hunk 9]
```diff
@@ -18,10 +18,10 @@ include("data_set_files.jl")
         all_sky(ds_sw; use_luts = true, λ_string = "sw")
     end
 
-    all_sky_pgp(ds_lw; use_luts = false, λ_string = "lw")
-    all_sky_pgp(ds_sw; use_luts = false, λ_string = "sw")
-    all_sky_pgp(ds_lw; use_luts = true, λ_string = "lw")
-    all_sky_pgp(ds_sw; use_luts = true, λ_string = "sw")
+    #    all_sky_pgp(ds_lw; use_luts = false, λ_string = "lw")
```
Comment (charleskawczynski) [inline test/test_allsky.jl:21]: Let's keep the PGP version working in case we need them.

**Comment by skandalaCLIMA** [comment]
@charleskawczynski : Please advise on modification of "mutable struct AtmosphericStatePGP{FT,I}" to add "param_set" field in AtmosphericStates.jl. The corresponding normal struct is "struct AtmosphericState{FT,I} "

**Comment by charleskawczynski** [comment]
> @charleskawczynski : Please advise on modification of "mutable struct AtmosphericStatePGP{FT,I}" to add "param_set" field in AtmosphericStates.jl. The corresponding normal struct is "struct AtmosphericState{FT,I} "

Just add a `param_set` to `AtmosphericStatePGP`, as you did to `AtmosphericState`.

**Comment by charleskawczynski** [comment]
You'll need to update the overloaded `convert` as well

**Comment by charleskawczynski** [comment]
Can we please not remove the PGP capability that was added?

**Comment by skandalaCLIMA** [comment]
> Can we please not remove the PGP capability that was added?

PGP code and tests are in place. Nothing is removed.

File: test/test_allsky.jl [hunk 10]
```diff
@@ -19,9 +19,9 @@ include("data_set_files.jl")
     end
 
     all_sky_pgp(ds_lw; use_luts = false, λ_string = "lw")
-    all_sky_pgp(ds_sw; use_luts = false, λ_string = "sw")
-    all_sky_pgp(ds_lw; use_luts = true, λ_string = "lw")
-    all_sky_pgp(ds_sw; use_luts = true, λ_string = "sw")
+    #    all_sky_pgp(ds_sw; use_luts = false, λ_string = "sw")
```
Comment (charleskawczynski) [inline test/test_allsky.jl:22]: Please Uncomment this

**Comment by charleskawczynski** [comment]
This PR drops coverage by 20% because the PGP kernels are not called. Also, they are slower because they’re not optimally called, there are a lot of conversions which are not reflected in the timing. This capability is useful as it decouples spatial discretization in the RRTMGP part of the code. I prefer we keep this capability in the code. The entire test suite is under 1 hr, so the additional testing time is not an issue.

File: test/test_allsky.jl (hunk 10 above)
Comment (skandalaCLIMA) [reply test/test_allsky.jl]: Tests uncommented.

File: test/test_rfmip_clear_sky_sw.jl [hunk 11]
```diff
@@ -15,8 +15,8 @@ include("data_set_files.jl")
         rfmip_clear_sky_sw(ds, TwoStream)
     end
 
-    rfmip_clear_sky_sw_pgp(ds, OneScalar)
-    rfmip_clear_sky_sw_pgp(ds, TwoStream)
+    #    rfmip_clear_sky_sw_pgp(ds, OneScalar)
```
Comment (charleskawczynski) [inline test/test_rfmip_clear_sky_sw.jl:18]: Please uncomment these as well

File: test/test_rfmip_clear_sky_sw.jl [hunk 12]
```diff
@@ -29,14 +29,14 @@ include("data_set_files.jl")
...
 
-    Δt_all[
-        "clear_sky_sw_pgp",
-        "1scl",
-    ] = @elapsed rfmip_clear_sky_sw_pgp(ds, OneScalar)
-    Δt_all[
-        "clear_sky_sw_pgp",
-        "2str",
-    ] = @elapsed rfmip_clear_sky_sw_pgp(ds, TwoStream)
+    #    Δt_all[
```
Comment (charleskawczynski) [inline test/test_rfmip_clear_sky_sw.jl:32]: Please uncomment these as well

**Pull request merged**